src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,52 @@
+package duke;
+
+import duke.command.Command;
+import duke.exceptions.DukeException;
+import duke.parser.Parser;
+import duke.task.TaskList;
+import duke.ui.Response;
+import duke.ui.Ui;
+
+import java.io.IOException;
+
+public class Duke {
+
+    private Storage store;
+    private TaskList taskList;
+    private Ui ui;
+    private Parser parser;
+
+    private Duke(Storage store, TaskList taskList, Ui ui, Parser parser) {
+        this.store = store;
+        this.taskList = taskList;
+        this.ui = ui;
+        this.parser = parser;
+    }
+
+    public static Duke startDuke() {
```

Comment:
This is an interesting approach to initializing Duke. Is there a reason why you decided to use a static method, `Duke.startDuke()` to start the program instead of `new Duke()`? Just curious!

Reply:
Thought it would be better to not include too much logic in the constructor, hence I moved it to a static method instead!

Reply:
Ahhh I see